### PR TITLE
Refine app typings and drop type casts

### PR DESCRIPTION
**Pull request by transphorm**
## Summary
- tighten button press and native component props with explicit React Native event and style types
- type NFC scan screen params and errors to remove `any`
- parse NFC responses using `unknown` inputs for safer handling

## Testing
- `yarn lint` *(fails: Unexpected any warnings)*
- `yarn build`
- `yarn workspace @selfxyz/contracts build` *(fails: Invalid account config)*
- `yarn types`
- `yarn workspace @selfxyz/common test`
- `yarn workspace @selfxyz/circuits test` *(fails: Unsupported signature algorithm: undefined)*
- `yarn workspace @selfxyz/mobile-app test`

------
https://chatgpt.com/codex/tasks/task_b_689502400190832d9e61bc9952bfeb63

## Summary by CodeRabbit

* **Refactor**
  * Improved type safety and clarity across the app by refining TypeScript type annotations for props, state, navigation, and utility functions.
  * Enhanced error handling by using safer error extraction patterns and stricter error types.
  * Added runtime checks and validation for certain user inputs and API responses.
  * Improved analytics event parameter handling to ensure only JSON-compatible data is sent.
  * Exported additional interfaces for broader usage across the app.

* **Tests**
  * Added comprehensive tests for analytics utilities, covering diverse and edge case scenarios.
  * Simplified and clarified existing test cases for better maintainability.

### CODE REVIEW

File: app/src/utils/cloudBackup/index.ts [hunk 1]
```diff
@@ -15,6 +15,14 @@ import {
 
 export const STORAGE_NAME = Platform.OS === 'ios' ? 'iCloud' : 'Google Drive';
 
+function isDriveFile(file: unknown): file is { id: string } {
```
Comment (transphorm) [inline app/src/utils/cloudBackup/index.ts:18]: new type guard function. pre-existing logic verifies that this still works

File: app/src/utils/analytics.ts [hunk 2]
```diff
@@ -36,43 +38,65 @@ export interface EventParams {
...
 }
 
 const segmentClient = createSegmentClient();
 
-function cleanParams(params: Record<string, any>) {
-  const newParams = {};
-  for (const key of Object.keys(params)) {
-    if (typeof params[key] !== 'function') {
-      (newParams as Record<string, any>)[key] = params[key];
+function coerceToJsonValue(value: unknown): JsonValue | undefined {
```
Comment (transphorm) [inline app/src/utils/analytics.ts]: new type guard logic

**Comment by transphorm** [comment]
@coderabbitai summary

**Comment by transphorm** [comment]
fyi @aaronmgdr `yarn types` passes locally
